packages/react-chain/src/render/RenderClient.tsx

```diff
@@ -1,18 +1,20 @@
-import * as ReactDOM from 'react-dom'
-import createContext from '../Context'
+import { render } from 'react-dom'
+import Session from '../Session'
 import { ReactChain } from '../ReactChain'
 
 function startClient(chain: ReactChain, domNode: Element) {
-  const context = createContext()
+  const session = new Session()
 
-  async function refresh(onComplete = () => {}) {
-    const element = await chain.getElement(context)
-    return await chain.renderClient(context, () => {
-      ReactDOM.render(element, domNode, onComplete)
+  async function refresh(onComplete?: Function) {
+    const element = await chain.getElement(session)
+
+    return await chain.renderBrowser(session, () => {
+      // Note: Typecasting to avoid weird, incorrect type checks.
+      render(element as any, domNode as any, onComplete as any)
```

Comment:
Just for the future: I wonder about this api, i.e. having `getElement`, `renderBrowser` and `renderServer`, and forcing the user to manage the `element`.

Maybe the render functions could call `getElement` and manage the `element`. I can think of one reason to still have a `getElement` outside rendering is some kind of preloading of other urls (not supported right now).